### PR TITLE
Fix Retain cycles, cleanup destinationViewController

### PR DESCRIPTION
**Pull request by sapzildj**
Thank you for great framework!
This is what I needed.

But there are some retain cycles.
And cleanup would be needed to be correct life cycle of destinationViewController.

Please check it.

### CODE REVIEW

File: JustPeek/Classes/PeekController.swift [hunk 1]
```diff
@@ -50,8 +50,10 @@ internal protocol PeekHandler {
     fileprivate var peekHandler: PeekHandler?
     fileprivate var sourceViewController: UIViewController?
     
+    public var nativeForceTouchDisabled = false
```
Comment (gianlucatranchedone-je) [inline JustPeek/Classes/PeekController.swift]: What is this for?

File: JustPeek/Classes/PeekNativeHandler.swift [hunk 2]
```diff
@@ -21,6 +23,11 @@ internal class PeekNativeHandler: NSObject, PeekHandler, UIViewControllerPreview
     
     func previewingContext(_ context: UIViewControllerPreviewing, viewControllerForLocation l: CGPoint) -> UIViewController? {
         peekContext?.destinationViewController = delegate?.peekContext(peekContext!, viewControllerForPeekingAt: l)
+        
+        if peekContext?.destinationViewController != nil {
+            let gesture = context.previewingGestureRecognizerForFailureRelationship
+            gesture.addObserver(self, forKeyPath: ObservingKeyPath, options: .new, context: nil)
```
Comment (gianlucatranchedone-je) [inline JustPeek/Classes/PeekNativeHandler.swift]: Why do we need to use KVO here?

File: JustPeek/Classes/PeekController.swift (hunk 1 above)
Comment (sapzildj) [inline JustPeek/Classes/PeekController.swift]: This is for using PeekReplacementHandler by force. I think that some users might be want it.

File: JustPeek/Classes/PeekNativeHandler.swift (hunk 2 above)
Comment (sapzildj) [inline JustPeek/Classes/PeekNativeHandler.swift]: In my case PeekNativeHandler's peekContext.destinationViewController should be nil when UIGestureRecognizerState is ended. It wasn't nil until another peek event is occured.

**Comment by federicocappelli-je** [comment]
Hi @sapzildj this library is now obsolete and not supported anymore, I'm going to close this PR.

**Pull request closed without merging**